drv/transceivers-server/src/main.rs

```diff
@@ -166,17 +165,16 @@ impl ServerImpl {
         if let Err(e) = self.leds.update_led_state(next_state) {
             ringbuf_entry!(Trace::LEDUpdateError(e));
         }
-        self.led_on = next_state;
-
         // handle system LED
-        match self.system_led_state {
-            LedState::On => self.system_led_on = true,
-            LedState::Blink => self.system_led_on = !self.system_led_on,
-            LedState::Off => self.system_led_on = false,
-        }
-        if let Err(e) = self.leds.update_system_led_state(self.system_led_on) {
+        if let Err(e) = match self.system_led_state {
+            LedState::On => self.leds.update_system_led_state(true),
+            LedState::Blink => self.leds.update_system_led_state(self.blink_on),
+            LedState::Off => self.leds.update_system_led_state(false),
+        } {
```

Comment:
Tiny nit - this might be slightly clearer if we broke it up into two steps; something like

```rust
let system_led_on = match self.system_led_state {
    LedState::On => true,
    LedState::Blink => self.blink_on,
    LedState::Off => false,
};
if let Err(e) = self.leds.update_system_led_state(system_led_on) {
```